fem/fe_coll.hpp

```diff
@@ -819,6 +819,20 @@ class NURBS_HDivFECollection : public NURBSFECollection
    virtual ~NURBS_HDivFECollection();
 };
 
+/** @brief Arbitrary order H(div) U H1  NURBS finite elements.
+    This class is identical to NURBS_HDivFECollection.
+    However fespace will behave slightly different for this FECollection,
```

Comment:
This class seems to be identical to `NURBS_HDivFECollection`, so how can there be different behavior for boundary DOFs?

Reply:
The logic that care of that was not copied correctly from my previous PR attempt.

There are some new subtle changed to the NURBSExtension routines that generate the HDiv/Hurcl extensions.
It has a flag that forces H1 boundaries if requested.

This flag is triggered by a `dynamic_cast` in the `UpdateNURBS()` routine of fespace.  So by defining another object the dynamic cast triggers the correct if.